lesson0.md

```diff
@@ -1,60 +1,106 @@
+
+1. Anyone has the freedom to run the software as they wish, for any purpose
+2. Anyone has the freedom to study how the program works, and change it as they wish
+3. Anyone has the freedom to redistribute the software
+4. Anyone has the freedom to distribute copies of their own modified version of the Software
+
+As a consequence, access to the code must be open and all 3rd party software needed to run it must also comply with the 4 fundamental freedoms.
+
+A few years later, in 1997, Eric Raymond wrote an essay called **"The Cathedral and the Bazaar"**, in which he described how the Linux operating system and the Free Software philosophy had created a surprisingly effective and collaborative way of writing software.
+
+This set the stage for the appearance, one year later, of the term **Open Source** software. Open Source software is a close relative of Free Software but with a stronger focus on the methodology rather than the philosophy.
+
+The movement has continued growing since then and nowadays there are many examples of popular Open Source software such as, for example, WordPress, Firefox, React, Python, the Apache web server and the MySQL database.
+
+> **Exercise:** Find out more examples of Open Source software.
+>
+> **Exercise:** Can you think of any software that is "free" (as in you pay £0.00 for it) but would not qualify as Open Source?
+
+If you wish to find out more on this topic, please watch the excellent documentary ["Revolution OS"](https://www.youtube.com/watch?v=4vW62KqKJ5A) on YouTube.
+
+
+This section has sourced information from:
+* [GNU - Free Software Definition](https://www.gnu.org/philosophy/free-sw.html)
+* [Make WordPress Training - What is Open Source](https://make.wordpress.org/training/handbook/user-lessons/what-is-open-source/)
+* [Opensource.com - What's the difference between open source software and free software?](https://opensource.com/article/17/11/open-source-or-free-software)
+
 
 ## What is WordPress? WordPress.com versus WordPress.org
 
-### What is WordPress
+WordPress is an Open Source Content Management System (CMS), released in 2003 by Matt Mullenberg and Mike Little. As of 2017, it has 59% of the CMS marketshare and 28% of all websites, worldwide.
+
+Although WordPress started as a blogging platform, it can also be used for building websites (large and small), e-commerce sites (with the WooCommerce plugin), private community platforms (with the BuddyPress plugin), forums (with the bbPress plugin), etc.
+
+Some examples of these could be: [Kinning Park Complex](http://www.kinningparkcomplex.org/), [Sony Music](https://www.sonymusic.com/), [Duracell Lighting](http://www.duracelllighting.com/), and [The Rolling Stones](http://www.rollingstones.com/)
 
-Include the following points:
+The introduction of the Rest API into WordPress core a couple of years ago, has also made it possible to write web or mobile apps that links into WordPress.
```

Comment:
Just thinking about how much "info" we're throwing at them at the start, I think this line about the REST API could go. We can cover the REST API in more detail later.